src/Refactoring-Core/RBInlineTemporaryRefactoring.class.st

```diff
@@ -38,55 +38,70 @@ RBInlineTemporaryRefactoring class >> model: aRBSmalltalk inline: anInterval fro
 ]
 
 { #category : 'preconditions' }
-RBInlineTemporaryRefactoring >> applicabilityPreconditions [
+RBInlineTemporaryRefactoring >> applicabilityPreconditions [ 
 
+	| validator |
+	
 	^ {
-		  (RBCondition definesSelector: selector in: class).
-		  (RBCondition withBlock: [
-			   self checkSelectedInterval.
-			   true ]) }
+		(ReDefinesSelectorsCondition new definesSelectors: (Array with: selector) in: class).
+		
+		(ReBlockCondition new 
+			block: [
+				self checkSelectedInterval.
+				validator := ReIsValidInstanceVariableName new.
+				(validator check: assignmentNode variable name)
+					ifFalse: [ 
+						self refactoringError: (validator violationMessageOn: assignmentNode variable name)].
+				true];
+			violatorErrorString: 'Check selected interval failed').		
+	}
 ]
 
 { #category : 'preconditions' }
 RBInlineTemporaryRefactoring >> checkSelectedInterval [
 
-	sourceTree := class parseTreeForSelector: selector.
-	sourceTree ifNil: [ self refactoringError: 'Could not parse source' ].
-	assignmentNode := sourceTree whichNodeIsContainedBy: sourceInterval.
-	assignmentNode isAssignment
-		ifFalse: [ self refactoringError: 'The selected node is not an assignment statement' ].
-	definingNode := assignmentNode whoDefines:
-		                assignmentNode variable name.
-	self hasOnlyOneAssignment
+	| searcher matchesCount |
+	searcher := self parseTreeSearcher.
+	matchesCount := 0.
+	searcher 
+		matches: assignmentNode variable name , ' := ``@object'
+		do: [ :aNode | matchesCount := matchesCount + 1 ].
+	(matchesCount = 1)
 		ifFalse: [ self refactoringError: 'There are multiple assignments to the variable' ].
-	( RBReadBeforeWrittenTester
+	
+	(RBReadBeforeWrittenTester
 		isVariable: assignmentNode variable name
-		writtenBeforeReadIn: definingNode )
-		ifFalse: [ self refactoringError: 'The variable is possible read before it is assigned' ]
+		writtenBeforeReadIn: definingNode)
+		ifFalse: [ self refactoringError: 'The variable is possibly read before it is assigned' ].
```

Comment:
Similar comment as above, could we just use the condition that checks if variable is read before written, and not check it by hand?